store/store.go

```diff
@@ -179,11 +179,11 @@ type Store struct {
 }
 
 func shouldRetryHttpResponse(attempt *retry.Attempt, resp *http.Response) bool {
-	return (resp.StatusCode == 500 || resp.StatusCode == 503) && attempt.More()
+	return (resp.StatusCode == 500 || resp.StatusCode == 503) && attempt.Next()
 }
 
 func shouldRetryError(attempt *retry.Attempt, err error) bool {
-	if !attempt.More() {
+	if !attempt.Next() {
```

Comment:
Next will block and wait even when the preceding conditions are false. It might be better to reorganize both of these functions as follows, so that Next is last:
```
if someCondition {
        return false
}
if otherConition {
        return false
}
return attempt.Next()
```